### PR TITLE
Onboarding notificationhubs for autogeneration

### PR DESCRIPTION
**Pull request by hunmix**
```
> azure-schema-generator@1.0.0 generate-single D:\code\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "notificationhubs/resource-manager"

[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "notificationhubs/resource-manager",
  "namespace": "Microsoft.NotificationHubs"
}
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\jlg87hvhbn --tag=all-api-versions --api-version=2014-09-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[8.55 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2014-09-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2014-09-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\yhbokr326aq --tag=all-api-versions --api-version=2016-03-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[11.79 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2016-03-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2016-03-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\hb21r5lko8n --tag=all-api-versions --api-version=2017-04-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[12.41 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2017-04-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2017-04-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
```

### CODE REVIEW

File: generator/whitelist.ts [hunk 1]
```diff
@@ -245,6 +245,11 @@ const whitelist: WhitelistConfig[] = [
         basePath: 'netapp/resource-manager',
         namespace: 'Microsoft.NetApp',
     },
+    {
+      basePath: 'notificationhubs/resource-manager',
+      namespace: 'Microsoft.NotificationHubs',
+      suffix: "namespaces_notificationhubs_authorizationRules"
```
Comment (anthony-c-martin) [inline generator/whitelist.ts]: Why include the suffix?

File: schemas/2019-04-01/deploymentTemplate.json [hunk 2]
```diff
@@ -144,6 +144,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01/Microsoft.Insights.json#/resourceDefinitions/components" },
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01/Microsoft.Insights.json#/resourceDefinitions/autoscalesettings" },
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01/Microsoft.Insights.json#/resourceDefinitions/webtests" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2014-02-26/microsoft.visualstudio.json#/resourceDefinitions/account" },
```
Comment (anthony-c-martin) [inline schemas/2019-04-01/deploymentTemplate.json]: Is this intentional?

**Comment by hunmix** [comment]
regenerate
```
> azure-schema-generator@1.0.0 generate-single D:\code\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "notificationhubs/resource-manager"

[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git fetch -q
[C:\Users\hunmix\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
Using whitelist config:
{
  "basePath": "notificationhubs/resource-manager",
  "namespace": "Microsoft.NotificationHubs"
}
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\obssg70547d --tag=all-api-versions --api-version=2014-09-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[8.3 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2014-09-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2014-09-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\llqexhdoeah --tag=all-api-versions --api-version=2016-03-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[7.98 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2016-03-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2016-03-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
[D:\code\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --version=3.0.6242 --use=@autorest/azureresourceschema@3.0.77 --azureresourceschema --output-folder=C:\Users\hunmix\AppData\Local\Temp\vbeyd10bcl --tag=all-api-versions --api-version=2017-04-01 --title=none --pass-thru:subset-reducer C:\Users\hunmix\AppData\Local\Temp\schm_azspc\specification\notificationhubs\resource-manager\readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\hunmix\.autorest\@autorest_core@3.0.6242\node_modules\@autorest\core\dist' (3.0.6242)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.77->3.0.77)
[9.01 s] Generation Complete
================================================================================================================================
Filename: D:\code\azure-resource-manager-schemas\schemas\2017-04-01\Microsoft.NotificationHubs.json
Provider Namespace: Microsoft.NotificationHubs
API Version: 2017-04-01
Resource Types (Resource Group Scope):
- namespaces
- namespaces/AuthorizationRules
- namespaces/notificationHubs
- namespaces/notificationHubs/AuthorizationRules
================================================================================================================================
```

File: schemas/2019-04-01/deploymentTemplate.json [hunk 3]
```diff
@@ -143,8 +143,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01/Microsoft.Insights.json#/resourceDefinitions/webtests" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-04-01/Microsoft.NotificationHubs.json#/resourceDefinitions/notificationHubs" },
```
Comment (anthony-c-martin) [inline schemas/2019-04-01/deploymentTemplate.json]: There shouldn't be any references to Microsoft.NotificationHubs in this file if they have been onboarded.

**Comment by anthony-c-martin** [comment]
Thanks!

File: schemas/2019-04-01/deploymentTemplate.json (hunk 3 above)
Comment (hunmix) [reply schemas/2019-04-01/deploymentTemplate.json]: Sorry, I have fixed that.

**Pull request merged**